HookCatcher/management/commands/newPRupdate.py

```diff
@@ -0,0 +1,85 @@
+import json
+import os
+
+import requests
+import yaml  # for parsing unicode json to strings
+from django.conf import settings  # database dir
+from django.core.management.base import BaseCommand, CommandError
+from HookCatcher.models import State
+
+STATES_FOLDER = 'states'  # folder within git repo that organizes the list of states
+# header of the git GET request
+GIT_HEADER = {
+    'Authorization': 'token ' + settings.GIT_OAUTH,
+}
+
+
+# function for accessing state JSON files and saving data into models
+def saveStates(gitBranch, gitPRnumber, gitCommit, gitSourceType):
+    # get the directory location of the states folder with the JSON states
+    gitContentURLtemp = os.path.join(settings.GIT_REPO_API, 'contents')
+    statesDir = STATES_FOLDER + '?ref=' + gitBranch
+    # example url https://api.github.com/repos/MingDai/kolibri/contents/states?ref=test-pr
+    gitContentURL = os.path.join(gitContentURLtemp, statesDir)
+    reqStatesDir = requests.get(gitContentURL, headers=GIT_HEADER)
+    jsonStatesDir = json.loads(reqStatesDir.text)
+
+    # save the json content for each file of the STATES_FOLDER
+    for eachState in jsonStatesDir:
+        # get the raw json file for each state
+        rawURL = eachState["download_url"]
+        reqRawState = requests.get(rawURL, headers=GIT_HEADER)
+        # save the json as a regular string rather than a unicode with yaml
+        jsonRawState = yaml.safe_load(reqRawState.text)
+
+        if str(gitSourceType).upper() == 'PR':
+            gitSourceName = gitPRnumber
+        else:
+            gitSourceName = gitBranch
+
+        s = State(state_name=jsonRawState['name'],
+                  state_desc=jsonRawState['description'],
+                  state_json=jsonRawState,
```

Comment:
should avoid dumping an object. If we did want to do this, we could use [`json.dumps`](https://docs.python.org/2/library/json.html) however as discussed, let's just output the URL.